lib/elixir/lib/task/supervised.ex

```diff
@@ -411,6 +411,19 @@ defmodule Task.Supervised do
         send(pid, {self(), {monitor_ref, spawned}})
         Map.put(waiting, spawned, {pid, :running})
 
+      {:max_children, ^monitor_ref} ->
+        stream_cleanup_inbox(monitor_pid, monitor_ref)
+        Process.demonitor(monitor_ref, [:flush])
```

Comment:
This is the thing I'm the least sure about. Basically, if the monitor process can't spawn a task because `:max_children`, it will kill all other running tasks and send `{:max_children, monitor_ref}` to the parent, then it will `exit(:normal)`. The only other cases it `exit(:normal)`s is when the parent process tells it to stop, but here we can't use `stream_close` (and tell the monitor process to stop) because we need to stop all processes and the monitor process right away. Demonitoring here works but I'm not sure it's what we want since we don't do it anywhere else. Definitely open to better solutions.

Reply:
Sounds like a better solution yes.

Reply:
@josevalim pushed a commit that fixes this. I think I did what you suggested. There's a bit of code repetition but it should work. Thoughts?